lib/irb/init.rb

```diff
@@ -446,13 +458,16 @@ def IRB.rc_file_generators
     end
     if home = ENV["HOME"]
       yield proc{|rc| home+"/.irb#{rc}"}
-      yield proc{|rc| home+"/.config/irb/irb#{rc}"}
+      if xdg_config_home.nil? || xdg_config_home.empty?
+        yield proc{|rc| home+"/.config/irb/irb#{rc}"}
+      end
     end
+  end
+
+  # possible irbrc files in current directory
+  def IRB.generate_current_dir_irbrc_files
```

Comment:
Let's make this private too.

Reply:
Done.
Also changed rc_file_generators to private too